### ISSUE TITLE
Remove all conjugated verbs from lexicon

### ISSUE THREAD

**Issue by dhowe** [dhowe/RiTa#357]
only for after removal of plurals is complete...

1. determine exactly how many conjugated verbs we are talking about?
2. identify a strategy for analysis of conjugated verbs no longer in lexicon
3. implement the strategy and identify any failing tests that it creates

**Comment by cqx931** [comment]
VB, base form
VBD, past tense
VBG, gerund or present participle
VBN, past participle
VBP, non-3rd person singular present
VBZ, 3rd person singular present

1. Delete vbd,vbg,vbn,vbp, vbz verbs that are not vb them self and at the same time has a vb in lexicon
Questions:
1.There are cases that some words only have vbn/vbd/vbg format, but no vb.
shall we create a new word in the form of vb and delete the other ones?
2.For conjugated verbs that are noun or adjective at the same time.("heated","crying"...) Are we going to delete them as well or not? If we delete them we can't generate those words out when we generate a noun/adjective.

2.Do you mean how to test whether we still have conjugated verbs in the lexicon or to analysis the deleted list? I am not very clear about what you want here.

**Comment by dhowe** [comment]
- For 1, we need an actual count of verbs (out of ~35,000 total) that we can remove. The simplest case are verbs that have a VB form and some other VB* form. The other two more complex cases are a) when there is no VB form, but only a VB* (here we should create a new VB), and b) when the VB* form is also some other POS, (a noun or adjective or whatever), in which case they cannot be removed, but the VB* form should be removed from the list.

- For 2, we need to come up with a mechanism to get the correct feature of analysis of sentences with conjugated verbs in them ("The dog ran after the boy."). The verb 'ran' will no longer be in the lexicon, but we still need the proper POS, phonemes, syllables, etc.

**Comment by cqx931** [comment]
1. update DictFromCMU.java in RiTa, get the count of verbs 
2.update rita_dict.js in JS folder
3.create Testcases (both in RiTaJS )
4.fix RiLexicon.js to deal with conjugated verbs
5. migrate things into Test cases and RiLexicon changes to RiTa.

**Comment by dhowe** [comment]
> 1. add DictWithoutConjugatedVerbs.java in RiTa, get the count of verbs   

This sounds like a good start, then we can see how many words we are talking about. Note that I wrote most (I think) of a program to handle this already, called DictFromCMU.java.

But lets discuss after step 1, before going on to any of the other steps

**Comment by cqx931** [comment]
yes, I see that file. I am writing my dict-generating code based on that file.(also the 1000 words generating code) But since that code is doing many things at the same time and we don't have the cmu file in src anymore, I think it is better to create a new file for conjugated verb(reusing related code in DictFromCMU.java )

**Comment by dhowe** [comment]
I've just re-checked in the syllable file to docgen, but feel free to, once you understand the code in DictFromCMU.java, to create a new file if you like...

**Comment by cqx931** [comment]
Here comes a first tryout:

Original entries : 29060
Matches : 5164
Deleted : 4509
Entries left : 24551

so the matches are those entries that have VB* and no VB, no other tags. They are all ok to delete or be replaced with VB in the future.
The deleted ones are those with corresponding 'VB' in the dictionary
I find it a bit complicated to get the right VB for the word, especially for non-regular ones.
Currently I used a quite simple filter to deal with 3rd-Person Present, 'ing' and 'ed'.
But more complicated ones could not be matched now.

<img width="659" alt="screen shot 2016-10-04 at 8 13 10 pm" src="https://cloud.githubusercontent.com/assets/3056154/19073727/1dd36d42-8a6f-11e6-9ecb-5a54a91ec404.png">

**Comment by cqx931** [comment]
btw, by deleting 4626 words, we already get the file size under 1MB.

**Comment by dhowe** [comment]
> btw, by deleting 4394 words, we already get the file size under 1MB.


what was it before?

**Comment by cqx931** [comment]
1.2MB

**Comment by dhowe** [comment]
on hold at least until plural nouns are fully removed
